File: patch_userid.js
const fs = require('fs');
let file = fs.readFileSync('src/components/Tabs/ProfileTab.jsx', 'utf8');

if (!file.includes("getUserId")) {
  file = file.replace("import { safeStorage } from '../../utils/safeStorage';", "import { safeStorage } from '../../utils/safeStorage';\nimport { getUserId } from '../../services/api';");
}

file = file.replace('userId={null}', 'userId={getUserId()}');

fs.writeFileSync('src/components/Tabs/ProfileTab.jsx', file);

File: patch_userid.py
with open('src/components/Tabs/ProfileTab.jsx', 'r') as f:
    content = f.read()

if "getUserId" not in content:
    content = content.replace("import { safeStorage } from '../../utils/safeStorage';", "import { safeStorage } from '../../utils/safeStorage';\nimport { getUserId } from '../../services/api';")

content = content.replace('userId={null}', 'userId={getUserId()}')

with open('src/components/Tabs/ProfileTab.jsx', 'w') as f:
    f.write(content)
